pkg/dmaasbackup/periodic_backup.go

```diff
@@ -130,6 +129,19 @@ func (d *dmaasBackup) cleanupPeriodicSchedule(dbkp *v1alpha1.DMaaSBackup) error
 		return nil
 	}
 
+	if !dbkp.Spec.PeriodicFullBackupCfg.DisableSuccessfulBackupRetain {
+		successfulScheduleIdx, err := d.getSuccessfulScheduleIdx(dbkp)
```

Comment:
I don't understand this logic

Reply:
ok.. below check is noticed later. Can we add few comments?

Reply:
ok